File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F04.P3.cpp
#include <array>
#include <cstdint>
#include <cstdio>
#include <memory>

struct Node {
    std::int32_t id;
};

bool probe() {
// BEGIN PROBE F04.P3
auto first = std::make_shared<Node>(5);
auto second = std::array{first}[0];
bool same = first == second;
return same;
// END PROBE F04.P3
}

int main() {
    std::printf("%d\n", probe());
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F04.P3.rs
use std::cell::RefCell;
use std::rc::Rc;

struct Node {
    id: i32,
}

fn probe() -> bool {
// BEGIN PROBE F04.P3
let first = Rc::new(RefCell::new(Node { id: 5 }));
let second = first.clone();
let same = Rc::ptr_eq(&first, &second);
same
// END PROBE F04.P3
}

fn main() {
    println!("{}", probe());
}
